CLN - Change string formatting in plotting

ohad83 opened the pull request:
Changed the string formatting in pandas/plotting/*.py to f-strings.

- [x] tests passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

ref #29547

ohad83 committed:
CLN - Changed to f-string formatting in pandas/plotting/*.py

ohad83 committed:
CLN - Changed another string formatting to f-string

topper-123 commented on pandas/plotting/_core.py:
```diff
@@ -796,14 +793,12 @@ def __call__(self, *args, **kwargs):
-                    msg = "{} requires either y column or 'subplots=True'"
-                    raise ValueError(msg.format(kind))
+                    msg = f"{kind} requires either y column or 'subplots=True'"
+                    raise ValueError(msg)
```
No need for an intermediate ``msg`` here, just put the message inside ValueError.

topper-123 commented on pandas/plotting/_matplotlib/boxplot.py:
```diff
@@ -75,8 +75,8 @@ def _validate_color_args(self):
-                            "key '{0}' "
-                            "The key must be either {1}".format(key, valid_keys)
+                            f"key '{key}' "
+                            f"The key must be either {valid_keys}"
```
These two should just be on the same line.  Also add "." after ``'{key}'``.

ohad83 replied:
Agreed, though I moved the key line to the previous line, makes it more readable.

topper-123 commented on pandas/plotting/_matplotlib/boxplot.py:
```diff
@@ -269,8 +269,8 @@ def _get_colors():
-                            "key '{0}' "
-                            "The key must be either {1}".format(key, valid_keys)
+                            f"key '{key}' "
+                            f"The key must be either {valid_keys}"
```
Same here

topper-123 commented on pandas/plotting/_matplotlib/timeseries.py:
```diff
@@ -307,7 +307,7 @@ def _maybe_convert_index(ax, data):
 
 def _format_coord(freq, t, y):
-    return "t = {0}  y = {1:8f}".format(Period(ordinal=int(t), freq=freq), y)
+    return f"t = {Period(ordinal=int(t), freq=freq)}  y = {y:8f}"
```
IMO there is too much packed in this f-string. Could you put the ``Period(..)`` part in a varable and call that in the f-string?

topper-123 reviewed (changes requested):
Looks good, just a few comments.

ohad83 committed:
CLN - Changed string formatting
Removed unnecessary intermediates and added needed intermediate for
clarification.

topper-123 commented:
Thanks, @ohad83!
